src/lib.rs

```diff
@@ -1,11 +1,38 @@
-use bitcoin::ScriptBuf as RustBitcoinScriptBuf;
+use bitcoin::Amount as BitcoinAmount;
+use bitcoin::ScriptBuf as BitcoinScriptBuf;
+
+use error::ParseAmountError;
+
+pub mod error;
+pub use bitcoin::amount::ParseAmountError as BitcoinParseAmountError;
+pub use bitcoin::Network;
+
+macro_rules! impl_from_core_type {
```

Comment:
Great stuff! Took me a minute to parse it but I think this is an example of a macro that's simple and worth having. It also ensures we reduce the number of places where these simple workflows are written. I'm fine with those as is, but how do you feel about renaming the field like so:

```rust
macro_rules! impl_from_core_type {
    ($core_type:ident, $ffi_type:ident) => {
        impl From<$core_type> for $ffi_type {
            fn from(core_type: $core_type) -> Self {
                $ffi_type(core_type)
            }
        }
    };
}
```
?

Reply:
Makes sense to me